docs-chef-io/content/supermarket/supermarket_upgrade.md

```diff
@@ -13,93 +13,163 @@ publishDate = 2022-01-03
+
+          For more information on upgrading using `pg_dumpall` see the PostgreSQL 13 documentation, section [18.6.1 Upgrading Data via pg_dumpall](https://www.postgresql.org/docs/13/upgrading.html).
+              
+            ```bash
+            sudo -u supermarket /opt/supermarket/embedded/bin/pg_dumpall -U supermarket -p 15432 > /tmp/supermarket-dump-archive.sql
+            ```
+     1. Copy the backup to a separate disk, one that is not connected to the Supermarket.
+### Vacuum the Database
+#### **Option 1**: Vacuum the PostgreSQL database
+  `vacuumdb --all --full` rewrites the entire contents of all tables into a disk files with no extra space, and returns unused space to the operating system.
+
+  Vacuum the database:
 
-Running Version| Upgrade Version | Supported Version
----------|----------|---------
- A1 | B1 | C1
- A2 | B2 | C2
- A3 | B3 | C3
-
-## Upgrade Supermarket
-
-### Database Preparation
-
-### Upgrade Steps
-
-## External PostgreSQL
-
-The External PostgreSQL upgrade steps are provided as a courtesy. It is the responsibility of the user to upgrade and maintain any External PostgreSQL configurations.
-
-Follow these steps to upgrade the PostgreSQL major version.
-
-## Upgrade Supermarket
-
-Do we need to upgrade Supermarket first?
-
-### Backup PostgreSQL Database
-
-1. Stop the Supermarket application
-
-```bash
-sudo supermarket-ctl stop
-```
+  ```bash
+  /opt/supermarket/embedded/bin/vacuumdb --all --full -p 15432
+  ```
 
-1. Start the Supermarket PostgreSQL service
+  For more information on upgrading using `vacuumdb` see the PostgreSQL 13 documentation for [vacuumdb](https://www.postgresql.org/docs/13/app-vacuumdb.html).
+#### **Option 2**: Vacuum and Analyze the PostgreSQL database
+  `vacuumdb --all --full --analyze` rewrites the entire contents of all tables into a disk files with no extra space, returns unused space to the operating system, and collects statistics about the database. It stores the results in the `pg_statistic` system catalog, which is useful for planning efficient queries. This command takes considerably more time to run.
 
-```bash
-sudo supermarket-ctl start postgresql
-```
+  Vacuume and analyze the database:
 
-1. Back up the PostgreSQL database
+  ```bash
+  /opt/supermarket/embedded/bin/vacuumdb --all --full --analyze -p 15432
+  ```
+### Backup the Cleaned Database (optional)
 
-```bash
-/opt/supermarket/embedded/bin/pg_dumpall -U supermarket -p 15432 > /tmp/supermarket-dump.sql
-```
+  This is an optional step for the sufficiently paranoid. Estimate the time needed for a second (the 'working') backup based as the same amount of time used in the first backup. The working backup provides a closer restore point than the archival copy if the next step of reindexing fails.
```

Comment:
Let's remove this section. I'm not sure it works.

Reply:
done